### PR TITLE
Fix page truncation (XML parsing)

### PR DESCRIPTION
**Pull request by philschatz**
rather than HTML (which causes page truncation).

Fixes https://github.com/Connexions/cnx-recipes/issues/655

There are still other places in this repo that are lenient about parsing (HTML, and using `recover=True`); this PR only fixes the immediate issue that was reported

### CODE REVIEW

File: cnxepub/tests/test_models.py [hunk 1]
```diff
@@ -134,6 +134,19 @@ def test_sanitize_xml(self):
...
...r, and the switch. (c) When the switch is moved to position <em data-effect="italics">B</em>, the circuit reduces to a simple series connection of the resistor, the capacitor, and the switch. The voltage source is removed from the circuit.</span></div></div>\n</section>\n\n\n\n<section data-depth="1" id="fs-id1172101790526"><h3 data-type="title">Charging a Capacitor</h3>\n<p id="fs-id1172097266009">We can use Kirchhoff&#8217;s loop rule to understand the charging of the capacitor. This results in the equation <math display="inline"><semantics><mrow><mrow><mi>&#949;</mi><mo>&#8722;</mo><msub><mi>V</mi><mi>R</mi></msub><mo>&#8722;</mo><msub><mi>V</mi><mi>c</mi></msub><mo>=</mo><mn>0</mn><mo>.</mo></mrow></mrow><annotation-xml encoding="MathML-Content"><mrow><mi>&#949;</mi><mo>&#8722;</mo><msub><mi>V</mi><mi>R</mi></msub><mo>&#8722;</mo><msub><mi>V</mi><mi>c</mi></msub><mo>=</mo><mn>0</mn><mo>.</mo></mrow></annotation-xml></semantics></math> This equation can be used to model the charge as a function of time as the capacitor charges. Capacitance is defined as <math display="inline"><semantics><mrow><mrow><mi>C</mi><mo>=</mo><mi>q</mi><mtext>/</mtext><mi>V</mi><mo>,</mo></mrow></mrow><
```
Comment (philschatz) [inline cnxepub/tests/test_models.py]: This print should be removed

File: cnxepub/tests/test_models.py [hunk 2]
```diff
@@ -134,6 +134,19 @@ def test_sanitize_xml(self):
...
...pace width="0.2em"/><msup><mrow><mn>10</mn></mrow><mrow><mn>&#8722;3</mn></mrow></msup><mspace width="0.2em"/><mtext>F</mtext></mrow><mo>)</mo></mrow><mrow><mo>(</mo><mrow><mn>10</mn><mspace width="0.2em"/><mo>&#215;</mo><mspace width="0.2em"/><msup><mrow><mn>10</mn></mrow><mn>3</mn></msup><mspace width="0.2em"/><mtext>&#937;</mtext></mrow><mo>)</mo></mrow><mtext>ln</mtext><mrow><mo>(</mo><mrow><mn>1</mn><mo>&#8722;</mo><mfrac><mrow><mn>10</mn><mspace width="0.2em"/><mtext>V</mtext></mrow><mrow><mn>12</mn><mspace width="0.2em"/><mtext>V</mtext></mrow></mfrac></mrow><mo>)</mo></mrow><mo>=</mo><mn>179.18</mn><mspace width="0.2em"/><mtext>s</mtext><mo>=</mo><mn>2.98</mn><mspace width="0.2em"/><mtext>min</mtext><mo>.</mo></mrow></annotation-xml></semantics></math></div>\n</div>\n<p id="fs-id1172099623126">The <em data-effect="italics">RC</em> circuit has thousands of uses and is a very important circuit to study. Not only can it be used to time circuits, it can also be used to filter out unwanted frequencies in a circuit and used in power supplies, like the one for your computer, to help turn ac voltage to dc voltage.</p>\n</section>\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n\n  \n  \n  </div>"""
```
Comment (philschatz) [inline cnxepub/tests/test_models.py]: These really long lines cause Travis to fail (the linter complains that they are too long)

File: cnxepub/tests/test_models.py [hunk 3]
```diff
@@ -134,6 +134,19 @@ def test_sanitize_xml(self):
         xml = target(content, recover=False)
         self.assertEqual(xml, '<img src="aaa_bbb.png"/>')
 
+    def test_sanitize_xml_not_html(self):
+        # self.maxDiff = None
```
Comment (philschatz) [inline cnxepub/tests/test_models.py]: This commented code should be removed

File: cnxepub/tests/test_models.py [hunk 4]
```diff
@@ -134,6 +134,22 @@ def test_sanitize_xml(self):
...
         self.assertEqual(xml, '<img src="aaa_bbb.png"/>')
 
+    def test_sanitize_xml_not_html(self):
+        # The serializer used lxml.html.create_fragments which caused truncation
+        from ..models import _sanitize_xml as target
+        test_dir = os.path.dirname(__file__)
+
+        source_file = open(os.path.join(test_dir, './test_models_source.xml'), 'r')
+        source = source_file.read().decode('utf-8')
+        source_file.close()
```
Comment (pumazi) [inline cnxepub/tests/test_models.py]: @philschatz, Future reference...

```python
with open('file.txt', 'r') as file:
    file.read()
```
the with block will auto close the file. But this works just fine.

Comment (philschatz) [reply cnxepub/tests/test_models.py]: Thanks! I knew about `with` but wasn't sure if it was a python3 thing or not. A quick search of how to read a file in 2.7 resulted in the 3-steps included in the PR. I'll update the PR 😄

**Pull request merged**